src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlySpanND.cs

```diff
@@ -0,0 +1,468 @@
+            if (array == null)
+            {
+                if (start != 0 || _linearLength != 0)
+                    ThrowHelper.ThrowArgumentOutOfRangeException();
+                this = default;
+                return; // returns default
+            }
+
+#if TARGET_64BIT
+            // See comment in SpanND<T>.Slice for how this works.
+            if ((ulong)(uint)start + (ulong)(uint)length > (ulong)(uint)array.Length)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+#else
+            if ((uint)start > (uint)array.Length || (uint)_linearLength > (uint)(array.Length - start))
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+#endif
+            if (_linearLength != SpanNDHelpers.CalculateTotalLength(lengths))
+                ThrowHelper.ThrowArgument_LengthsMustEqualArrayLength();
+
+            _reference = ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(array), (nint)(uint)start /* force zero-extension */);
+
+            _lengths = lengths.ToArray().AsSpan();
+            _strides = SpanNDHelpers.CalculateStrides(Rank, lengths);
+        }
+
+        /// <summary>
+        /// Creates a new read-only span over the target unmanaged buffer.  Clearly this
+        /// is quite dangerous, because we are creating arbitrarily typed T's
+        /// out of a void*-typed block of memory.  And the length is not checked.
+        /// But if this creation is correct, then all subsequent uses are correct.
+        /// </summary>
+        /// <param name="pointer">An unmanaged pointer to memory.</param>
+        /// <param name="lengths">The <see cref="System.ReadOnlySpan{T}"/> with the lengths of each dimension.</param>
+        /// <param name="isPinned">If the underlying data is pinned.</param>
+        /// <param name="strides">The <see cref="System.ReadOnlySpan{T}"/> with the lengths of each stride.</param>
+        /// <exception cref="ArgumentException">
+        /// Thrown when <typeparamref name="T"/> is reference type or contains pointers and hence cannot be stored in unmanaged memory.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the specified length is negative.
+        /// </exception>
+        [CLSCompliant(false)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe ReadOnlySpanND(void* pointer, ReadOnlySpan<nint> lengths, bool isPinned, ReadOnlySpan<nint> strides = default)
+        {
+            _linearLength = SpanNDHelpers.CalculateTotalLength(lengths);
+
+            if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
+                ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));
+            if (_linearLength < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+            if (_linearLength != SpanNDHelpers.CalculateTotalLength(lengths))
+                ThrowHelper.ThrowArgument_LengthsMustEqualArrayLength();
```

Comment:
This is repeating the logic already done

Reply:
Fixed.